test/mocks/loggers.cpp

```diff
@@ -1 +1,17 @@
 #include "loggers.h"
+
+#include <cstddef>
+#include <ostream>
+
+std::ostream& operator<<(std::ostream& stream,
+                         const std::vector<MockLogger::Entry>& entries) {
+  stream << "<BEGIN " << entries.size() << " LOG ENTRIES>";
+  for (std::size_t i = 0; i < entries.size(); ++i) {
+    const auto& entry = entries[i];
+    const auto kind_name =
+        entry.kind == MockLogger::Entry::ERROR ? "ERROR" : "STARTUP";
+    stream << '\n' << (i + 1) << ". " << kind_name << ": ";
```

Comment:
prefixing `\n` instead of suffixing it seems a bit odd, is there a corner-case driving that?

Reply:
I don't see the difference. I'll suffix them instead.

Reply:
I think what I was going for is "newlines only if there are entries," but this code doesn't achieve that. I'll fix it.